boardswarm/src/hifive_p550_mcu.rs

```diff
@@ -0,0 +1,404 @@
+    });
+}
+
+enum ChannelMessage {
+    SomPower(SomPowerParameters),
+    BootSel(BootSelParameters),
+}
+
+impl McuCommand {
+    fn to_channel_message(
+        self,
+        parameters: Box<dyn erased_serde::Deserializer<'static> + Send>,
+    ) -> Result<ChannelMessage, erased_serde::Error> {
+        let message = match self {
+            McuCommand::HifiveP550McuSomPower => {
+                let parameters = SomPowerParameters::deserialize(parameters)?;
+                ChannelMessage::SomPower(parameters)
+            }
+            McuCommand::HifiveP550McuBootSel => {
+                let parameters = BootSelParameters::deserialize(parameters)?;
+                ChannelMessage::BootSel(parameters)
+            }
+        };
+        Ok(message)
+    }
+}
+
+async fn process_som_power(
+    reader: &mut BufReader<ReadHalf<SerialStream>>,
+    write: &mut WriteHalf<SerialStream>,
+    parameters: &SomPowerParameters,
+) -> Result<(), McuError> {
+    let (value, message) = parameters.mode.to_status();
+
+    mcu_wait_prompt(reader, write).await?;
+    mcu_send(write, "sompower-g\n").await?;
+
+    let response = Regex::new(r"^Som Power Status: (?<mode>.*)$").unwrap();
+    let ret = mcu_wait_response(reader, &response, MCU_RESPONSE_TIMEOUT).await?;
+    let caps = response.captures(&ret).ok_or(McuError::ResponseParse)?;
+    if caps["mode"].to_lowercase() == parameters.mode.as_ref() {
+        return Ok(());
+    }
+
+    let request = format!("sompower-s {}\n", value);
+    mcu_send(write, &request).await?;
+    let response = Regex::new(&message).unwrap();
+    mcu_wait_response(reader, &response, MCU_RESPONSE_TIMEOUT).await?;
+
+    Ok(())
+}
+
+async fn process_boot_sel(
+    reader: &mut BufReader<ReadHalf<SerialStream>>,
+    write: &mut WriteHalf<SerialStream>,
+    parameters: &BootSelParameters,
+) -> Result<(), McuError> {
+    let (mode, value, bits) = parameters.mode.to_status();
+
+    mcu_wait_prompt(reader, write).await?;
+    mcu_send(write, "bootsel-g\n").await?;
+
+    let response =
+        Regex::new(r"^Get: Bootsel Controlled by: (?<mode>.*), bootsel\[3 2 1 0\]:(?<bits>.*)$")
+            .unwrap();
+    let ret = mcu_wait_response(reader, &response, MCU_RESPONSE_TIMEOUT).await?;
+    let caps = response.captures(&ret).ok_or(McuError::ResponseParse)?;
+    if caps["mode"] == mode && bits.as_ref().is_none_or(|b| caps["bits"] == *b) {
+        return Ok(());
+    }
+
+    let request = format!("bootsel-s {} 0x{:x}\n", mode, value);
+    mcu_send(write, &request).await?;
+
+    let response =
+        Regex::new(r"^Set: Bootsel Controlled by: (?<mode>.*), bootsel\[3 2 1 0\]:(?<bits>.*)$")
+            .unwrap();
+    let ret = mcu_wait_response(reader, &response, MCU_RESPONSE_TIMEOUT).await?;
+    let caps = response.captures(&ret).ok_or(McuError::ResponseParse)?;
+    if caps["mode"] == mode && bits.as_ref().is_none_or(|b| caps["bits"] == *b) {
+        return Ok(());
+    }
+
+    Err(McuError::ResponseParse)
```

Comment:
fwiw slightly more idiomatic
```suggestion
    if caps["mode"] == mode && bits.as_ref().is_none_or(|b| caps["bits"] == *b) {
        Ok(())
    } else {
      Err(McuError::ResponseParse)
    }
```